pex/resolver.py

```diff
@@ -56,6 +58,10 @@ def merge(self, other):
 
 
 class _ResolvableSet(object):
+  @classmethod
```

Comment:
I just generally always use classmethods instead of staticmethods to allow for future subclassing/testing.  For example, we could write an implementation like so:

```python
class BuggyResolvableSet(_ResolvableSet):
  @classmethod
  def normalize(cls, name):
    return name
```

and then assert that it fails when running the same test suite as the regular implementation.

As it stands, we're subclassing it nowhere and it's an internal class so it doesn't really give us any leverage over a staticmethod but why be inconsistent.